src/org/traccar/FilterHandler.java

```diff
@@ -84,7 +84,12 @@ public FilterHandler() {
     }
 
     private boolean filterInvalid(Position position) {
-        return filterInvalid && !position.getValid();
```

Comment:
I think you can still use return expression straight away. You just need to groups the rest in parentheses.

Also, I liked the split of latitude and longitude checks on separate lines as you had before. I think it makes it more readable.